easysid.cpp

```diff
@@ -152,42 +152,66 @@ EXPORT void set_filter_res(uint8_t res)
     sid.write(23, reg);
 }
 
-EXPORT void set_filter_enable(uint8_t voice)
+EXPORT void set_filter(uint8_t voice, bool enable)
 {
-    if (voice < 1 || voice > 3) {
+    if (voice < 0 || voice > 2) {
         return;
     }
     char reg = sid.read_state().sid_register[23];
-    reg &= ~0x0f;
-    reg |= 1 < (voice - 1);
+    reg &= ~(1 << voice) & 0x0f;
```

Comment:
I'm pretty sure this doesn't do what you want it to do. Check the precedence of `~` and `&`.